### ISSUE TITLE
Kitchen sink app for a11y testing

### ISSUE THREAD

**Issue by hippee-lee** [vmware/clarity#1314]
# Create a demo app for a11y testing
We need to repurpose some of the development demos so they can be packaged up on kitchensink apps and published to a gh-pages branch on one of the core member forks. This needs to be accessible (pun intended) for a third party to go through the components and test them.

## General
    1. Most of our demos seem like they can be ported as is
    2. There are 30 kitchen sink pages that need to be created
    3. There are between 80~100 demos to be ported (and some might need functionality added) depending on how we cound the things like button variations
    4. One option is to use the sample-app and hook up the publishing into that since it already builds

## Estimated work
What needs to happen to turn the demo app into a kitchensink app that can be hosted on gh-pages? This is only focused on our Angular component demos. `* - these demos have a laot of variations but should be copy/paste into a kitchen sink page`
1. Build process modification
    - Modify existing build? 
    - Build it with webpack?
    - Does it need aot? 
    - publishing shouldn't be to hard as we already have a model with gh-pages in our documentation branch
3. Create kitchen sink pages (# is estimated demos to port)
    - Alerts (1*):
        - styles
        - dismissable
        - alert w/ action buttons
        - all level alert with actions and dismissable
    - Badges (1*):
        - color options
        - statuses
        - There are 14 different colors/statuses for badges
        - No functionality associated 
    - Button Group (1*): 
        - Types (13): Primary (3), Outline (3), Flat (1), Inverse (1), Mixed buttons (2), Small buttons (3)
            - Icons variations - is it something they will test? 
        - Checkboxes (1): did not seem accessible
        - Radios (1): did not seem accessible
        - Cards (2): examples use checkbox and radio examnples
        - Overflow behavior (1): Use the Overflow toggle demo w/ the overflow behavior active 
        - The types demos are all static demos and may need functionlaity added
    - Buttons (1*)
        - (72 combos) Primary, secondary and tertiary (flat) buttons with the following colors
            - primary, success, warning, danger?
            - small sizes
            - block sizes
            - each combo has a disabled state
            - inverse: seems to be only outline
        - Toggles (4)
            - active and on + disabled for each state
    - Cards (3)
        - With action
        - Clickable (on hover)
        - W/ dropdown action
    - Checkboxes (4) 
        - unchecked
        - checked
        - diabled + unchecked
        - disabled + checked
    - Color (3)
        - Direct port of the existing code to standalone page
        - luminance
        - Contrast
    - Datagrid (2*)
        - Like Full demo with selectable and multiline and pagination in the footer
        - Add in hidable columns
        - Add in expandable rows
        - Scrolling ?? (horizontal and vertical)
    - Dropdown (1*)
        - Primary, success, warning danger?
            solid, outline and button link
        - Icon Toggle
        - Clarity Icon toggle
        - Nested menu
        - In a header
    - Forms (2)
        - With validation - note that tool tips in our demos did not seem accessible.
        - Compact
        - Grid
        - Note - not sure how complex this demo needs to be for a11y. The Add employee Address example with above variations might be alright
    - Iconography (??)
        - Not sure, are there visibility considerations they may test here?
       - Might be able to test with use elsewhere
    - Input Fields (8)
        - text, password, textarea, url, tel, date, time, datetime-local
    - Labels (1*)
        - colors: default, purple, blue, orange, light-blue (label-1, label-2, label-3, label-4, label-5)
        - statuses: info, success, warning, danger (error)
        - clickable
        - With badges
    - Lists (1*)
        - unstyled, ordered, mixed style, compact, list in cards
    - Login (0)
        - this isn't really a component but it incorporates typography, inputs, checkbox, (what might be an alert in red), button button-link
    - Modal (4)
        - Sizxes: sm, lg, xl
        - With close icon cancel/ok actions
        - Without close icon and close action
        - With input
    - Navigation (2)
        - Tabs
        - Vertical Nav (Includes header)
    - Progress Bars (1*)
        - Colors: normal, success, danger/warning
        - Animations - do these get a11y tested? - assuming no
        - Sidebar
        - Cards
        - Inline: simple, complex layout, stacked, stakced + resized
    - Radio Buttons (1)
        - might be able to combine with forms
    - Selects (1)
        - might be able to combine with forms
    - Signposts (1)
        - known issue for active outline not showing #1274
    - Spinners (6)
        - Default, inline and dark background
        - Sizes: sm, md, lg
    - Stack View (1)
        - Basic Stack view - did not seem a11y
    - Tables (6)
        - basic: multiline, no-border, compact, compact+no-border
        - vertical: compact, no-border
    - Tabs (1)
    - Tree View (1)
        - New Tree View demo
    - Toggles (4)
        - Seems same as in buttons (duplicate?)
        - States: off+enabled, off+disabled, on+enabled, on+disabled
        - w/ and w/o labels
    - Tooltips (1)
        - Will this be better tested inside a more compelxt component like forms or modal? 
    - Typography (1)
        - Weights: 200, 400, 500, 600 
        - Headers: h1, h2, h3, h4, h5, h6
        - Body text and paragraphs: Body text, section header, table, grid, form text, form label, column header, table footer, chart data, small headers, tags, labels, badges
        - Might be able to combine this for some things like labels, tags, badges etc
    - Wizard (4)
        - Basic (close icon, cancel, next buttons)
        - Basic + form (with validation)
        - Basic w/ header actions
        - Do we want to include inline?

**Comment by mathisscott** [comment]
Yes, we want to include inline wizard for testing/accessibility purposes.

**Comment by mathisscott** [comment]
```
Feedback on the approach - 
1) use sample app 
2) add kitchen sink pages with components needed for a11y testing
3) add a publish task to our build process that will publish to any gh-pages branch *except vmware/clarity*
```

Yup. Yup. And yup.
